### ISSUE TITLE
CKV2_AZURE_31 False positive

### ISSUE THREAD

**Issue by askpatrickw** [bridgecrewio/checkov#5111]
**Describe the issue**
Not sure why this is failing since we have a network security group defined and associated.

**Examples**
```terraform
resource "azurerm_subnet" "app-snt-cms" {
  name                 = "${local.environment}-snt-${local.region}-${local.application}-${local.incremental}"
  resource_group_name  = azurerm_resource_group.app-rsg-cms.name
  virtual_network_name = azurerm_virtual_network.app-vnt-cms.name
  address_prefixes     = ["10.0.1.0/24"]
}

resource "azurerm_network_security_group" "app-nsg-cms" {
  name                = "${local.environment}-nsg-${local.region}-${local.application}-${local.incremental}"
  resource_group_name = azurerm_resource_group.app-rsg-cms.name
}
resource "azurerm_subnet_network_security_group_association" "app-nsgxsnt-cms" {
  subnet_id                 = azurerm_virtual_network.app-vnt-cms.subnet[0].id
  network_security_group_id = azurerm_network_security_group.app-nsg-cms.id
}
```

**Version (please complete the following information):**
checkov: 2.3.247

**Additional context**

https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/graph_checks/azure/AzureSubnetConfigWithNSG.yaml

**Comment by gruebel** [comment]
hey @askpatrickw thanks for reaching out.

The check fails, because there is no direct connection between the `azurerm_subnet` and `azurerm_subnet_network_security_group_association`. you are referencing the subnet through the VNET resource, which is quite odd `azurerm_virtual_network.app-vnt-cms.subnet[0].id`. You just need to change to the actual subnet resource and then it will pass

```hcl
resource "azurerm_subnet_network_security_group_association" "app-nsgxsnt-cms" {
  subnet_id                 = azurerm_subnet.app-snt-cms.id
  network_security_group_id = azurerm_network_security_group.app-nsg-cms.id
}
```

**Comment by askpatrickw** [comment]
Thanks for your help!
